src/core_write.cpp

```diff
@@ -231,8 +231,27 @@ void TxToUniv(const CTransaction& tx, const uint256& hashBlock, bool include_add
             in.pushKV("txinwitness", txinwitness);
         }
         if (calculate_fee) {
-            const CTxOut& prev_txout = txundo->vprevout[i].out;
+            const Coin& prev_coin = txundo->vprevout[i];
+            const CTxOut& prev_txout = prev_coin.out;
+
             amt_total_in += prev_txout.nValue;
+            switch (verbosity) {
+                case TxVerbosity::SHOW_TXID:
+                case TxVerbosity::SHOW_DETAILS:
+                    break;
+
+                case TxVerbosity::SHOW_DETAILS_AND_PREVOUT:
+                    UniValue p(UniValue::VOBJ);
+                    UniValue o_script_pub_key(UniValue::VOBJ);
+
+                    p.pushKV("value", ValueFromAmount(prev_txout.nValue));
+                    p.pushKV("height", uint64_t(prev_coin.nHeight));
+                    p.pushKV("generated", bool(prev_coin.fCoinBase));
+                    ScriptPubKeyToUniv(prev_txout.scriptPubKey, o_script_pub_key, true, true);
```

Comment:
The convention seems to be:

```suggestion
                    ScriptPubKeyToUniv(prev_txout.scriptPubKey, o_script_pub_key, /* fIncludeHex */ true, /* include_addresses */ true);
```